Add in LICENSE.

clalancette opened the pull request:
Fixes #45

clalancette committed:
Add in LICENSE.

Signed-off-by: Chris Lalancette <clalancette@gmail.com>

clalancette commented:
Pulls: https://github.com/ros-visualization/rqt_publisher/pull/46
Gist: https://gist.githubusercontent.com/clalancette/372b70eb21c4c4f89e0d7fe1c7dae45b/raw/c0a099cba53d9dc8088bbaca2480db30e3b9ce03/ros2.repos
BUILD args:  --packages-above-and-dependencies rqt_publisher
TEST args:  --packages-above rqt_publisher
ROS Distro: rolling
Job: ci_launcher
ci_launcher ran: https://ci.ros2.org/job/ci_launcher/14642
* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=21925)](http://ci.ros2.org/job/ci_linux/21925/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=16192)](http://ci.ros2.org/job/ci_linux-aarch64/16192/)
* Linux-rhel [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-rhel&build=1439)](http://ci.ros2.org/job/ci_linux-rhel/1439/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=22651)](http://ci.ros2.org/job/ci_windows/22651/)

tfoote commented on test/test_copyright.py:
```diff
@@ -0,0 +1,23 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
```
Why 2019 for a new file?

clalancette replied:
Hm, good question.  Well, it was copied from an existing file, which is why I left it as 2019.  But I could change that.

tfoote commented on test/test_copyright.py:
```diff
@@ -0,0 +1,23 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
```
The package is under BSD but this license here is Apache

clalancette replied:
Good point.  Since this file is copied from another place (I think I copied this from https://github.com/ros2/ros2cli/blob/rolling/ros2action/test/test_copyright.py), should we change the license on this file to be BSD, or should we add in Apache-2.0 as one of the licenses on this package?

tfoote replied:
It would be much cleaner to keep the existing license and not cause dual licensing woes. In the case the file is bordering on de minimus. As it's 7 total lines of code, two are imports, three are matching a defined prototype with annotations thus no possible expressiveness. And that leaves two lines with one function call, saving it's return value and asserting. Which has very little expressiveness. For good measure you could re-implement those 2 lines and then it being a new composition, apply the project's license.

clalancette replied:
All right, done in 7dd18c5.

clalancette committed:
Update from review.

Change the license of the test file itself to BSD,
also make small improvement to the test_copyright.py.

Signed-off-by: Chris Lalancette <clalancette@gmail.com>

ahcorde commented:
* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=22237)](http://ci.ros2.org/job/ci_linux/22237/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=16497)](http://ci.ros2.org/job/ci_linux-aarch64/16497/)
* Linux-rhel [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-rhel&build=1759)](http://ci.ros2.org/job/ci_linux-rhel/1759/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=23003)](http://ci.ros2.org/job/ci_windows/23003/)
